docs/advanced_topics.rst

```diff
@@ -0,0 +1,43 @@
+Advanced topics
+=================
+
+In this section, you can find some advanced functionalities and topics that are just partially mentioned in the API description. For the typical use case, DALI default configuration should perform well out of the box and you don't need to worry about those.
+
+DALI internal data format
+-------------------------
+
+Internally, DALI processes data in NHWC format. To reshape it to NCHW please use ``\*Permute`` operator family. It also means that if mentioned operators are used in the middle of the pipeline then next operations will take the wrong assumption about processed data format and the result is undefined.
+
+Thread affinity
+---------------
+
+The rationale behind this functionality it to allow pining DALI threads to given CPU. It can save the overhead of worker threads jumping from core to core and increase performance for CPU heave workloads.
```

Comment:
The purpose of this functionality is to allow pinning DALI threads to the given CPU. Thread affinity can save the overhead of worker threads jumping from core to core. Hence this functionality can increase the performance for CPU heave workloads.

Reply:
Adjusted